lib/wam.coffee

```diff
@@ -237,23 +236,26 @@ downloadAddon = (addonURL, addonIdentifier) ->
     tmpFilePath = path.resolve os.tmpdir(), "#{addonIdentifier}-#{filename}.zip"
 
     log.info "Downloading #{addonIdentifier.magenta}..."
-
-    request {
-      followAllRedirects: true,
-      url: "#{addonURL}"
-    }
-    .pipe fs.createWriteStream "#{tmpFilePath}"
-    .on 'error', (error) ->
-      console.log error
-      if error.code is ErrorCodeDoesntExist
-        reject "Error writing Addon #{addonIdentifier.yellow} in temporary
-                folder #{tmpFilePath.yellow}."
-      else
+    
+    cs
+      method: 'GET'
+      uri: addonURL
+      rejectUnauthorized: false
+      realEncoding: null
+    .then (body) ->
+
+      buffer = Buffer.from body, 'utf8'
+
+      fs.writeFile tmpFilePath, buffer, (error) ->
```

Comment:
Having the body already in memory, is it worth creating a buffer instead of writing it directly? Previous implementation took advantage of `request` returning a stream and piping it directly to the file system

Reply:
Cloudscraper didn't provide interface to take advantage of stream read/write, any request will be converted to string internally. I have to convert body back to buffer so that the zip format won't be corrupted.